question24/main.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const agreementLinkBtn = document.getElementById('js-agreement-link');
+const modal = document.getElementById('js-modal');
+const modalClose = document.getElementById('js-close');
+const mask = document.getElementById('js-mask');
+const checkbox = document.getElementById('js-checkbox');
+const submitButton = document.getElementById('js-submit-button');
+const target = document.querySelector(".last_content");
+
+
+agreementLinkBtn.addEventListener('click',()=>{
+    modal.classList.remove('hidden');
+    mask.classList.remove('hidden');
+});
+modalClose.addEventListener('click',()=>{
+    modal.classList.add('hidden');
+    mask.classList.add('hidden');
+});
+mask.addEventListener('click',()=>{
+    modalClose.click();
+});
+
+const options = {
+  root: document.getElementById('js-modal'),
+  threshold: 1
+};
+
+const checkWhenIntersect = (entries) => {
+  entries.forEach(entry => {
```

Comment:
IMO
I will share what I learned from sae-san!!
[Destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) are available.
This is because there is only one target element.
```suggestion
const checkWhenIntersect = ([entry]) => {
if (entry.isIntersecting) {
      checkbox.checked = true;
      checkbox.disabled = false;
}
```

Reply:
Thank you for your review! Thanks for the nice idea! It's so much smerter this way! Fixed so please confirm.

fixed commit : https://github.com/csakyo/homework/pull/38/commits/c531ef0e70585f0f3c1020068e87f5e2f839ee25